cargo-pgrx/src/command/sudo_install.rs

```diff
@@ -0,0 +1,92 @@
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+use owo_colors::OwoColorize;
+
+use crate::command::package::Package;
+use crate::CommandExecute;
+
+/// Like `cargo pgrx install`, but uses `sudo` to copy the extension files
+#[derive(clap::Args, Debug, Clone)]
+#[clap(author)]
+pub(crate) struct SudoInstall {
+    /// Package to build (see `cargo help pkgid`)
+    #[clap(long, short)]
+    package: Option<String>,
+    /// Path to Cargo.toml
+    #[clap(long, value_parser)]
+    manifest_path: Option<PathBuf>,
+    /// Compile for release mode (default is debug)
+    #[clap(long, short)]
+    release: bool,
+    /// Specific profile to use (conflicts with `--debug`)
+    #[clap(long)]
+    profile: Option<String>,
+    /// Build in test mode (for `cargo pgrx test`)
+    #[clap(long)]
+    test: bool,
+    /// The `pg_config` path (default is first in $PATH)
+    #[clap(long, short = 'c', value_parser)]
+    pg_config: Option<PathBuf>,
+    /// The directory to output the package (default is `./target/[debug|release]/extname-pgXX/`)
+    #[clap(long, value_parser)]
+    out_dir: Option<PathBuf>,
+    #[clap(flatten)]
+    features: clap_cargo::Features,
+    #[clap(from_global, action = ArgAction::Count)]
+    verbose: u8,
+}
+
+impl From<SudoInstall> for Package {
+    fn from(value: SudoInstall) -> Self {
+        Package {
+            package: value.package,
+            manifest_path: value.manifest_path,
+            debug: !value.release,
+            profile: value.profile,
+            test: value.test,
+            pg_config: value.pg_config,
+            out_dir: value.out_dir,
+            features: value.features,
+            verbose: value.verbose,
+        }
+    }
+}
+
+impl CommandExecute for SudoInstall {
+    fn execute(self) -> eyre::Result<()> {
+        fn make_absolute(p: &Path) -> PathBuf {
+            PathBuf::from_iter(vec![PathBuf::from("/").as_path(), p])
```

Comment:
Wow does this ever look wrong (it's not wrong for this use case ofc, but still)

Reply:
Yeah why don't you just use `std::fs::canonicalize` together with `shellexpand` as needed?

Reply:
```rust
let src = src.canonicalize()?;
let dest = make_absolute(src.strip_prefix(&outdir)?).canonicalize()?;
```

The `.src.strip_prefix(&outdir)` call here includes the leading `/`.  So without `make_absolute()` the path ends up being relative to the current working directory.  Ask me how I know.